ompi/runtime/ompi_spc.c

```diff
@@ -0,0 +1,281 @@
+    /* The data structure has been allocated, so we simply initialize all of the counters
+     * with their names and an initial count of 0.
+     */
+    for(i = 0; i < OMPI_NUM_COUNTERS; i++){
+        events[i].name = counter_names[i];
+        events[i].value = 0;
+    }
+}
+
+/* Initializes the SPC data structures and registers all counters as MPI_T pvars.
+ * Turns on only the counters that were specified in the mpi_spc_attach MCA parameter.  
+ */
+void ompi_spc_init()
+{
+    int i, j, ret, found = 0, all_on = 0;
+
+    /* Initialize the clock frequency variable as the CPU's frequency in MHz */
+    sys_clock_freq_mhz = opal_timer_base_get_freq() / 1000000;
+
+    events_init();
+
+    /* Get the MCA params string of counters to turn on */
+    char **arg_strings = opal_argv_split(ompi_mpi_spc_attach_string, ',');
+    int num_args      = opal_argv_count(arg_strings);
+
+    /* If there is only one argument and it is 'all', then all counters
+     * should be turned on.  If the size is 0, then no counters will be enabled.
+     */
+    if(num_args == 1){
+        if(strcmp(arg_strings[0], "all") == 0)
+            all_on = 1;
+    }
+
+    /* Turn on only the counters that were specified in the MCA parameter */
+    for(i = 0; i < OMPI_NUM_COUNTERS; i++){
+        if(all_on)
+            attached_event[i] = 1;
+        else{
+            /* Note: If no arguments were given, this will be skipped */
+            for(j = 0; j < num_args && found < num_args; j++){
+                if(strcmp(counter_names[i], arg_strings[j]) == 0){
+                    attached_event[i] = 1;
+                    found++;
+                }
+            }
+        }
+
+        /* ########################################################################
+         * ################## Add Timer-Based Counter Enums Here ##################
+         * ########################################################################
+         */
+        /* If this is a timer event, sent the corresponding timer_event entry to 1 */
+        if(i == OMPI_MATCH_TIME || i == OMPI_OOS_MATCH_TIME)
+            timer_event[i] = 1;
+        else
+            timer_event[i] = 0;
+
+        /* Registers the current counter as an MPI_T pvar regardless of whether it's been turned on or not */
+        ret = mca_base_pvar_register("ompi", "runtime", "spc", counter_names[i], counter_descriptions[i],
+                                     OPAL_INFO_LVL_4, MPI_T_PVAR_CLASS_SIZE,
+                                     MCA_BASE_VAR_TYPE_UNSIGNED_LONG_LONG, NULL, MPI_T_BIND_NO_OBJECT,
+                                     MCA_BASE_PVAR_FLAG_READONLY | MCA_BASE_PVAR_FLAG_CONTINUOUS,
+                                     ompi_spc_get_count, NULL, ompi_spc_notify, NULL);
+
+        /* Initialize the mpi_t_indices array with the MPI_T indices.
+         * The array index indicates the SPC index, while the value indicates
+         * the MPI_T index.
+         */
+        if(ret != OPAL_ERROR){
```

Comment:
The coding convention for if is
`if(condition) {`
apply everywhere.

Reply:
Ok, I've gone through and changed this for all of my ifs.  I assume this is also the case for loops?  So I should have:
for(stuff) {
instead of:
for(stuff){